src/lib.rs

```diff
@@ -596,6 +601,7 @@ mod op {
             (Value::Int(a), Value::Int(b)) => Value::Int(a + b),
             (Value::String(a), Value::String(b)) => Value::String(Rc::from(format!("{}{}", a, b))),
             (Value::Tuple(a), Value::Tuple(b)) if a.len() == b.len() => tuple_bin_op(a, b, add),
+            (Value::Unknown, a) | (a, Value::Unknown) => a.clone(),
```

Comment:
Swallowing a Value::Unknown this way is a bit sketchy. Ideally we'd only swallow when type checking but that's potentially _a lot_ of branching. `#[cfg(debug)]`?

Reply:
A nice fix here would be to pass in the tuple (a, a) for checking with this operation, so we don't have to assume the type supports the operation. And then only have the case `(Value::Unknown, Value::Unknown) => Value::Unknown`. Maybe?